config/routes.rb

```diff
@@ -2,5 +2,10 @@
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
-  # root "articles#index"
+  # root "users#index"
+  get "/users", to: "users#index"
+  get "/users/:id", to: "users#show"
+  get "/users/:id/posts", to: "posts#index"
+  get "/users/:id/posts/:post_id", to: "posts#show"
```

Comment:
- [ ] [OPTIONAL] Good job adding the required routes 👏 However, this approach usually takes up too many lines, clutters your route file, and makes it hard to understand. I would suggest you do something along the lines of 👇 
```ruby
resources :users, only: [:index, :show] do
  resources :posts, only: [:index, :show]
end

```

Reply:
Thanks @codepantha, for your time and your feedback, I will change the code according to your suggestions.